addons/mail/static/src/discuss/core/web/discuss_sidebar_categories.js

```diff
@@ -84,6 +84,7 @@ export class DiscussSidebarCategories extends Component {
         return category.threads.filter((thread) => {
             return (
                 (thread.is_pinned || thread.group_based_subscription) &&
+                !thread.readonly &&
```

Comment:
Ok for this part, but I think readonly thread currently being viewed should still appear on the sidebar.